File: test/fixtures/benchmarks/python/duplications.py
# Two near-identical helpers — jscpd should detect this clone with
# default thresholds (--min-lines 5 --min-tokens 50). Phase 10i.0.3:
# per-language duplication fixture. Bodies are sized comfortably
# above the 50-token threshold so jscpd's tokenizer reliably matches
# regardless of small whitespace differences across runs.


def summarize_items_a(items):
    total = 0
    sum_positive = 0
    sum_negative = 0
    count_positive = 0
    count_negative = 0
    for item in items:
        if item > 0:
            total = total + item
            sum_positive = sum_positive + item
            count_positive = count_positive + 1
        else:
            total = total - item
            sum_negative = sum_negative + item
            count_negative = count_negative + 1
    avg_pos = sum_positive / count_positive if count_positive > 0 else 0
    avg_neg = sum_negative / count_negative if count_negative > 0 else 0
    return total + avg_pos + avg_neg


def summarize_items_b(items):
    total = 0
    sum_positive = 0
    sum_negative = 0
    count_positive = 0
    count_negative = 0
    for item in items:
        if item > 0:
            total = total + item
            sum_positive = sum_positive + item
            count_positive = count_positive + 1
        else:
            total = total - item
            sum_negative = sum_negative + item
            count_negative = count_negative + 1
    avg_pos = sum_positive / count_positive if count_positive > 0 else 0
    avg_neg = sum_negative / count_negative if count_negative > 0 else 0
    return total + avg_pos + avg_neg

File: test/fixtures/benchmarks/java/Duplications.java
// Phase 10k.1.0 — per-language duplication fixture, Java row.
//
// Two near-identical methods — jscpd (language-agnostic) should detect
// this clone with default thresholds (--min-lines 5 --min-tokens 50).
// Bodies are sized comfortably above the 50-token threshold so the
// match is reliable across whitespace differences. Mirrors the
// kotlin/python duplications fixture layout.

public class Duplications {
    public double summarizeItemsA(int[] items) {
        int total = 0;
        int sumPositive = 0;
        int sumNegative = 0;
        int countPositive = 0;
        int countNegative = 0;
        for (int item : items) {
            if (item > 0) {
                total = total + item;
                sumPositive = sumPositive + item;
                countPositive = countPositive + 1;
            } else {
                total = total - item;
                sumNegative = sumNegative + item;
                countNegative = countNegative + 1;
            }
        }
        double avgPos = countPositive > 0 ? (double) sumPositive / countPositive : 0.0;
        double avgNeg = countNegative > 0 ? (double) sumNegative / countNegative : 0.0;
        return total + avgPos + avgNeg;
    }

    public double summarizeItemsB(int[] items) {
        int total = 0;
        int sumPositive = 0;
        int sumNegative = 0;
        int countPositive = 0;
        int countNegative = 0;
        for (int item : items) {
            if (item > 0) {
                total = total + item;
                sumPositive = sumPositive + item;
                countPositive = countPositive + 1;
            } else {
                total = total - item;
                sumNegative = sumNegative + item;
                countNegative = countNegative + 1;
            }
        }
        double avgPos = countPositive > 0 ? (double) sumPositive / countPositive : 0.0;
        double avgNeg = countNegative > 0 ? (double) sumNegative / countNegative : 0.0;
        return total + avgPos + avgNeg;
    }
}
